src/openrct2/windows/top_toolbar.c

```diff
@@ -361,6 +364,12 @@ static void window_top_toolbar_mouseup(rct_window *w, sint32 widgetIndex)
 	case WIDX_NEWS:
 		window_news_open();
 		break;
+	case WIDX_MUTE:
+		if(gGameSoundsOff)
+			audio_unpause_sounds();
+		else
+			audio_pause_sounds();
+		break;
```

Comment:
`sound_enabled` only controls sound effects, not ride music. So you would also need to do `ride_music_enabled`. You still need to call `audio_pause_sounds` otherwise you will continue to hear the crowd noise and other various things.